File: advent-of-code-2021/day1/Solution.java
import java.io.*;
import java.util.*;

public class Solution {
    public static void main(String[] args) throws Exception {
        File file = new File("E:\\Coding\\advent-of-code-2021\\day1\\input.txt");
        BufferedReader br = new BufferedReader(new FileReader(file));
        ArrayList<Integer> nums = new ArrayList<Integer>();
        int answer = 0;
        int c = 0;
        int lastSum = 0;

        String s;
        while ((s = br.readLine()) != null) {
            int curr = Integer.parseInt(s);

            if (nums.size() == 3) {
                nums.set(c % 3, curr);
                int secondSum = nums.get(0) + nums.get(1) + nums.get(2);
                if (secondSum > lastSum) {
                    answer++;
                }

                lastSum = secondSum;
            } else {
                nums.add(curr);
            }

            c++;
        }

        System.out.println(answer);
        br.close();
    }
}

File: advent-of-code-2021/day1/solution.py
def main():
    print("Part one: ", part_one())
    print("Part two: ", part_two())

def part_one():
    answer = 0; prev = 0

    with open("input.txt") as file:
        for line in file:
            curr = int(line)
            if (prev != 0 and curr > prev):
                answer += 1
            prev = curr

    return answer

def part_two():
    nums = []
    answer = 0; c = 0; prevSum = 0

    with open("input.txt") as file:
        for line in file:
            curr = int(line)

            if (len(nums) == 3):
                nums[c % 3] = curr
                sum = nums[0] + nums[1] + nums[2]

                if (sum > prevSum):
                    answer += 1
                
                prevSum = sum
            else:
                nums.append(curr)

            c += 1

    return answer


if __name__ == "__main__":
    main()
